### ISSUE TITLE
Script causes tumblr to fail to load any posts

### ISSUE THREAD

**Issue by ProtoDot** [enchanted-sword/dashboard-unfucker#27]
### Expected behavior
Old tumblr layout is used to load posts as expected
### Actual behavior
Old tumblr layout is used, but fails to load posts or recognize that I'm logged in'
### Repro steps

1. Install and enable script
2. Go to tumblr.com
3. "Server turned to dust" or something along those lines is displayed. A green log in button is shown where the user profile would be found.

**Comment by enchanted-sword** [comment]
This is seemingly a Tumblr issue, I'd been running into it occasionally for months before I ever made this script. It's not consistent and I have no idea what causes it, but you can just reload the page to get rid of it.

**Comment by enchanted-sword** [comment]
Fixed in latest commit

**Comment by hmmmmmmm** [comment]
I'm on 3.0.3 and this is still happening intermittently, as well as the script just sometimes not loading when returning to the dash from search/other pages.

**Comment by enchanted-sword** [comment]
> I'm on 3.0.3 and this is still happening intermittently, as well as the script just sometimes not loading when returning to the dash from search/other pages.

Are you currently using Twilight Sparkle's poll unlocker userscript by any chance? The cause for the crash as far as we know is a script attempting to define `window.___INITIAL_STATE___` after it has already been created and instead setting its value to `undefined` while Tumblr's source JS is attempting to read it. I fixed the issue in my script by always un-defining the window property after the page finishes loading so that on each successive load my script attaches the getter and setter _before_ Tumblr accesses it. I only discovered this yesterday, and Twilight's poll unlocker is still prone to causing the crash.

It could also possibly be related to window load times, but as of v3.0.3 I can no longer replicate it on my machine so I have no way of telling.

**Comment by enchanted-sword** [comment]
> I fixed the issue in my script by always un-defining the window property after the page finishes loading so that on each successive load my script attaches the getter and setter before Tumblr accesses it.

Load in this case meaning either navigating between Tumblr pages via anchors or the use of the `window.tumblr.navigate()` property, or soft reloading the page with Ctrl + Shift + R

**Comment by hmmmmmmm** [comment]
I'm not using Twilight's poll script, so this is purely the dash unfucker (maybe an interaction with XKit Rewritten?).

It's fine on the types of loading you mentioned, but it seems to be triggered on loading the page using a bookmark. If it's in a fresh tab, it works properly, but if Tumblr is already open in the same tab, it will either cause this behaviour (dash with unfucker adjustments but no posts and login button) or similar behaviour where the vanilla dash loads but some elements are hidden (radar etc).

I'm using Firefox 116.0.2 and Tampermonkey 4.19.0 for reference.

**Comment by enchanted-sword** [comment]
I started testing it more and it started crashing again for me for no discernible reason. Even weirder, sometimes after crashing it would un-crash itself. I think what's happening is that the setter is being added _after_ the property has been set but _before_ it gets called, so the getter always returns `undefined`. I toyed around with a few different solutions, but I realized that in this weird load case where the script is executed in between the property being set and fetched I can instead just add a getter that instead just fetches a proxy object. It could be prettier, but I can't get it to crash any more even when opening it via a bookmark.

Unrelated: when only the radar and stuff were hidden, that's probably because of XKit having an identical feature. v3.0.3 skipped modifying the DOM completely when loading into a page that still had the vertical nav feature enabled.
